components/prism-core.js

```diff
@@ -236,21 +236,24 @@ var _ = {
 	highlightAllUnder: function(container, async, callback) {
 		var env = {
 			callback: callback,
+			container: container,
 			selector: 'code[class*="language-"], [class*="language-"] code, code[class*="lang-"], [class*="lang-"] code'
 		};
 
 		_.hooks.run('before-highlightall', env);
 
-		var elements = container.querySelectorAll(env.selector);
+		env.elements = Array.prototype.slice.apply(container.querySelectorAll(env.selector));
+
+		_.hooks.run('before-highlightall-name-pending', env);
```

Comment:
`after-elements-query`?

Reply:
Still want that `all` in there.

`after-highlightall-elements-query`?

(Seems like [Java enterprise](https://projects.haykranen.nl/java/) has left its marks on me.)

Reply:
Technically, the hook is in `highlightUnder`... `after-highlight-elements-query` would work for me. I don't think it needs the `all`, tbh, as it's not specific to the plugin, so doesn't need the name to match the same way.

Reply:
It's in `highlightAllUnder`. Also, we have the method `highlightElement` which makes the name `after-highlight-elements-query` just a tad confusing IMO.

Instead, we could choose a name that describes that this hook is the last one executed before all elements are iteratively highlighted.

`before-all-elements` ? (but this is kinda missing the verb/action that is done afterward)

Reply:
`before-all-elements-highlighted`? Naming things is hard...

Reply:
Let's just roll with this for now. 
~Nothing is more permanent than a temporary solution.~ Maybe we'll find something better later on.

Reply:
`*-highlighted` or `*-highlight` ?

Reply:
`*-highlight` is more in line with the naming conventions of other hooks. None of them use past tense.